Cleanup node cloning and skip sandbox

IDisposable opened the pull request:
Root node is detected when `cloneNode` is passed `null` for  `parentComputedStyles`
Hoisted `sandbox` up higher
Don't ever clone the `sandbox` node
Remove spurious `util.delay(0)` in rendering chain
Removed extra `cloneChildrenInOrder` method and in-lined the body into `cloneChildren`
Made the references to the `original` element and its children and styles more apparent.

IDisposable committed:
Clean up root handling and skip sandbox

Root is detected when `cloneNode` is passed `null` for  `parentComputedStyles`
Remove spurious `util.delay(0) ` in rendering chain
Hoisted `sandbox` up higher and don't ever clone the sandbox

IDisposable committed:
Clarify cloning of children

Removed extra method and made the references to the original element and its children and styles more apparent.

IDisposable commented:
(sorry, fixed the base to be the 3.0 lineage.

IDisposable committed:
Work on table padding issue #95

Changed `getDefaultStyle` to create the element in the sandbox so it is pure (not tainted by CSS / ShadowDOM in the main document).
Never clone SCRIPT elements (that's just asking for trouble).
Removed `imagediff` package as it's not useful anymore.

IDisposable commented:
I will likely rebase this once I get the v2 - dynasty branch stuff correct and make `master` the current branch for v3

IDisposable commented on src/dom-to-image-more.js:
```diff
@@ -206,7 +205,6 @@
         options = options || {};
         return toSvg(domNode, options)
             .then(util.makeImage)
-            .then(util.delay(0))
```
This was put in a VERY long time ago to fix some yielding issues. It needs to go away.

IDisposable commented on src/dom-to-image-more.js:
```diff
@@ -1018,7 +1007,8 @@
         // Ensure that there is some content, so that properties like margin are applied.
-        defaultElement.textContent = '.';
+        // we use zero-width space to handle FireFox adding a pixel
+        defaultElement.textContent = '\0x200B';
```
Not sure if this is better or worse, but as long as we're not trying to inject `SCRIPT` tags, it shouldn't matter anymore :)

meche-gh replied:
To get the string working in my console:
```suggestion
        defaultElement.textContent = '\u200b';
```

Generally I think the ZWSP is better:
- we're now thinking about "empty" elements for `getDefaultStyle`
- the hack now narrowly targets the bug
- plus for free, we avoid a value collision when an element's CSS has the precise width of a `.` but is empty

IDisposable replied:
Applied this in another branch, thanks!

IDisposable committed:
Update src/dom-to-image-more.js

Co-authored-by: meche-gh <114668551+meche-gh@users.noreply.github.com>

IDisposable committed:
Add ability to insert the cloned node into the test

The allows inspecting the properties of the clone.

IDisposable committed:
Simplified the SVG

Removed the setting of the `top` and `left`
Set the `width` and `height` on the `foreignObject` element to the target size.
Removed setting of `width` and `height` on the `svg` element.

IDisposable committed:
Copy the size from the original object when possible

Falls back to the old `scrollWidth`/`scrollHeight` logic if the `width` and `height` aren't set in `px`.

IDisposable committed:
Revert to have dimensions on SVG object.

IDisposable committed:
Work on table padding issue #95 (#104)

* Work on table padding issue #95

Changed `getDefaultStyle` to create the element in the sandbox so it is pure (not tainted by CSS / ShadowDOM in the main document).
Never clone SCRIPT elements (that's just asking for trouble).
Removed `imagediff` package as it's not useful anymore.

* Add ability to insert the cloned node into the test

The allows inspecting the properties of the clone.

* Simplified the SVG

Removed the setting of the `top` and `left`
Set the `width` and `height` on the `foreignObject` element to the target size.
Removed setting of `width` and `height` on the `svg` element.

* Copy the size from the original object when possible

Falls back to the old `scrollWidth`/`scrollHeight` logic if the `width` and `height` aren't set in `px`.

* Revert to have `width` and `height` dimensions on SVG object.

IDisposable committed:
Merge branch 'cleanup-clone' into master

IDisposable committed:
Merge pull request #107 from 1904labs/master

Catch up to master's clone-cleanups

IDisposable committed:
Fix merge of  "origin/cleanup-clone"

IDisposable commented:
Merged by hand.
